test/extended/testdata/deployments/deployment-hook-timeout.yaml

```diff
@@ -0,0 +1,57 @@
+apiVersion: v1
+kind: DeploymentConfig
+metadata:
+  creationTimestamp: null
+  generation: 12
+  labels:
+    run: example
+  name: example
+spec:
+  replicas: 1
+  selector:
+    run: example
+  strategy:
+    customParams:
+      command:
+      - openshift-deploy
+      - --loglevel=5
+    resources: {}
+    recreateParams:
+      intervalSeconds: 1
+      post:
+        execNewPod:
+          command:
+          - sleep
+          - "1"
+          containerName: example
+        failurePolicy: Ignore
+      pre:
+        execNewPod:
+          command:
+          - sleep
+          - "1"
+          containerName: example
+        failurePolicy: Ignore
+      timeoutSeconds: 600
+      updatePeriodSeconds: 1
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        run: example
+    spec:
+      containers:
+      - command:
+        - /bin/bash
+        - -c
+        - "sleep 1 && exit 0"
```

Comment:
Actually it would be really interesting to have such a test for MinReadySeconds but that's a different issue.